### ISSUE TITLE
Single source of truth for environment variable reference

### ISSUE THREAD

**Issue by jwulf** [camunda/camunda#19140]
**Is your feature request related to a problem? Please describe.**
I'm frustrated when I want to configure something via an environment variable, or implement an SDK feature that uses the same environment variable as Zeebe and I try to find out what features are configurable via environment variable and what the environment variable name is.

**Describe the solution you'd like**
I would like a single source of complete truth about the environment variables that Zeebe supports - their names, possible values, and effects.

**Describe alternatives you've considered**
Something that is programmatically generated would be the most easily maintained. Could environment variable configuration be coalesced as a concern into a single point in the codebase, optionally with extraction for documentation. If it were a single file similar to the proto file, which contains all the information in a single place with documentation in comments then it could be linked to from documentation, even if it were not extracted into the docs. 

**Additional context**
As an example, in the JS SDK, I put all the environment variable configuration in a single place to give users a reference: https://github.com/camunda/camunda-8-js-sdk/blob/main/src/lib/Configuration.ts

**Comment by npepinpe** [comment]
ZDP Triage:
- Removing from our board, as this is a pure client issue.

**Comment by jwulf** [comment]
This is a request to make it standard. Let me put the request this way: 

Please export all configuration environment variables with a description of their function, either as a single file in the code base, or a single file exported from the codebase programatically for consumption by documentation. 

Once we do this in every project that is configured via environment variables, we will have 100% documentation coverage, and in a consolidated fashion.

**Comment by akeller** [comment]
@npepinpe @korthout can you re-evaluate this with @jwulf's message? ☝️ We can absolutely consume a single file of env vars in the docs.

**Comment by korthout** [comment]
After talking to @jwulf, I'd like to recap what we discussed:
- This concerns a holistic solution to generate a list of all configurable env vars in C8 projects, not just for the Zeebe clients
- Planning should go through PDP

**Comment by akeller** [comment]
Raised here to get support to move into PDP - https://camunda.slack.com/archives/C02K3MV8X16/p1731097014382629

**Comment by akeller** [comment]
Feature requested via ProductBoard.

**Comment by jwulf** [comment]
We can generate documentation for configuration automatically from Spring Configuration annotations at least for the Spring SDK. 

See here for how this done: https://github.com/camunda/camunda/issues/28653

Here is an example of config generated from the source code: https://gist.github.com/jwulf/66906acc0efa320910df8c0c2d0166ae

This gives us complete coverage of Spring configuration properties. 

What's needed there is: 
1. Populate the description of the property via a comment on the property. The text should be generated by the engineer implementing the feature at the time that they write the annotation. This is the maximal point of context and knowledge. The description should be written with the end-user in mind, knowing that this will go into end-user documentation in a configuration table.
2. The description should include the equivalent environment variable name. 

Can we adopt the same approach for other components? If we can, then we have a solution that is amenable to automation.

**Comment by jonathanlukas** [comment]
Yes. Every spring boot application can apply a processor to generate a spring configuration properties json.

@tmetzke we already discussed this: generated documentation.

**Comment by megglos** [comment]
I lack a bit to understand the scope of this, @jwulf did you mainly mean the Spring SDK for which a reference is needed, so this not about the configuration of the Orchestration cluster components but solely the SDK?

I would take this with CamundaEx, my current understanding of addressing this is:
* have a single holistic reference of all properties supported by the SDK in the docs
* this should also include a differentiation of which of these properties are deprecated and their successor
* we need that for all currently supported minor version

In contrast to the status quo, where we have config snippets for certain scenarios, functionals or setups like here https://docs.camunda.io/docs/apis-tools/spring-zeebe-sdk/getting-started/#self-managed

**Comment by jonathanlukas** [comment]
I added this as a hackday project: https://confluence.camunda.com/spaces/camBPM/pages/274825445/Projects+Guidelines+2025#:~:text=Generated%20reference%20for%20the%20Spring%20SDK

**Comment by jwulf** [comment]
This is extractable from the engine with Spring introspection. 

There are multiple profiles and combinations of profiles possible.

Three formats: 

application.properties, yaml, and process env.

**Comment by jonathanlukas** [comment]
I am implementing 2 of the formats in our [properties reference](https://docs.camunda.io/docs/next/apis-tools/spring-zeebe-sdk/properties-reference/) as of now:

* environment variable
* property -> can be mapped to yaml 1:1, I see no need in adding a yaml snippet explict (if required, we could also generate it)

The view could be improved as the level of detail is really verbose as of now.

They are a complete listing of all properties that actually exist and is generated as you proposed.

On top, there is a configuration guide that explains the `camunda.client.mode` and the `camunda.client.auth.method` in detail.
